orderer/common/multichannel/registrar.go

```diff
@@ -577,8 +577,71 @@ func (r *Registrar) joinAsFollower(ledgerRes *ledgerResources, clusterConsenter
 // RemoveChannel instructs the orderer to remove a channel.
 // Depending on the removeStorage parameter, the storage resources are either removed or archived.
 func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
 	if r.SystemChannelID() != "" && channelID != r.SystemChannelID() {
 		return types.ErrSystemChannelExists
 	}
-	return errors.New("Not implemented yet")
+
+	cs, ok := r.chains[channelID]
```

Comment:
Add a check right above here that rejects the removal of the system channel, and returns "not yet implemented". Removal of the system channel is a different (Jira) story that requires a special process, since it mandates the conversion of every chain life-cycle from etcdraft+inactive to etcdraft+follower. This may require a restart of the orderer or the restart of every chain, as we'll decide when we tackle that story.

Reply:
Done